server/src/main/java/org/elasticsearch/action/admin/indices/mapping/put/PutMappingRequest.java

```diff
@@ -80,7 +80,9 @@ public PutMappingRequest(StreamInput in) throws IOException {
         super(in);
         indices = in.readStringArray();
         indicesOptions = IndicesOptions.readIndicesOptions(in);
-        type = in.readOptionalString();
+        if (in.getVersion().before(Version.V_8_0_0)) {
+            in.readOptionalString();
```

Comment:
We could check the type and throw an `IllegalArgumentException` here as we've been doing in other places.

Reply:
++